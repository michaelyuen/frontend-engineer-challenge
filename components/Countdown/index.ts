export * from "./Countdown";
export * from "./styles";
