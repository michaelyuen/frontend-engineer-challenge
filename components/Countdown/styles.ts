import styled from "styled-components";
import { breakpoints } from "../../styles/breakpoints";

export const CountdownContainer = styled.div`
  display: flex;
  flex-direction: column;

  h1,
  p {
    text-align: center;
  }

  h1 {
    font-size: 3em;
    margin: 0;

    @media (min-width: ${breakpoints.md}) {
      font-size: 4em;
    }
  }

  p {
    margin-bottom: 2em;
  }
`;

export const NumbersContainer = styled.div`
  display: grid;
  gap: 1em;
  grid-template-columns: repeat(3, minmax(50px, 150px));

  @media (min-width: ${breakpoints.lg}) {
    grid-template-columns: unset;
    grid-auto-columns: 150px;
    grid-auto-flow: column;
  }

  margin-bottom: 2em;
`;

export const Warning = styled.p`
  background: var(--warning-color);
  border-radius: var(--border-radius);
  color: black;
  padding: 1em;
`;
