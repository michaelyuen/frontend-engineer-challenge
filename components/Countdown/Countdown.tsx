import { StyledCountdown } from "./styles";

export const Countdown = () => {
  return (
    <StyledCountdown>
      <img src="koala-logo.png" />
      <h1>Koala Frontend Engineer Coding Challenge</h1>
      <p>Build a countdown clock!</p>
      <div
        style={{
          border: ".5px solid gray",
          width: 50,
          borderRadius: "3px",
          background: "#EFEFEF",
          padding: 3,
        }}
      >
        <div onClick={() => alert("TODO")}>Start</div>
      </div>
    </StyledCountdown>
  );
};
