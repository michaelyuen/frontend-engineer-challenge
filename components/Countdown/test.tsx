import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import { Countdown } from "./Countdown";

describe("Countdown", () => {
  test("renders", () => {
    render(
      <Countdown
        deadlineName="Take Home Test"
        deadlineValue="2022-10-17T08:00"
        onReset={jest.fn}
      />
    );
    expect(
      screen.getByText("Time until Monday, October 17th, 2022 at 8:00 AM")
    ).toBeInTheDocument();
  });

  /**
   * Not a good test unless it's updated to be stable in some way.
   * Currently the snapshot will change every time (as expected)
   * because the time until the deadline keeps changing
   */
  xtest("matches snapshot for a deadline in the future", () => {
    const { container } = render(
      <Countdown
        deadlineName="Take Home Test"
        deadlineValue="2023-10-17T08:00"
        onReset={jest.fn}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("matches snapshot for a deadline in the past", () => {
    const { container } = render(
      <Countdown
        deadlineName="Take Home Test"
        deadlineValue="2021-10-17T08:00"
        onReset={jest.fn}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("has no a11y violations", async () => {
    const { container } = render(
      <Countdown
        deadlineName="Take Home Test"
        deadlineValue="2022-10-17T08:00"
        onReset={jest.fn}
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});
