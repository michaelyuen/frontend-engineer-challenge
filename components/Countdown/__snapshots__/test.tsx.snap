// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown matches snapshot for a deadline in the past 1`] = `
<div>
  <div
    class="sc-bcXHqe ktLLPh"
  >
    <h1>
      Take Home Test
    </h1>
    <p>
      Time until Sunday, October 17th, 2021 at 8:00 AM
    </p>
    <div
      class="sc-gswNZR gYyeuC"
    >
      <div
        class="sc-jSUZER czOrtm"
      >
        <div
          class="sc-gKPRtg bPprQx"
        >
          <div
            class="top next"
          >
            0
          </div>
          <div
            class="flipper"
          >
            <div
              class="bottom"
            >
              0
            </div>
            <div
              class="top"
            >
              0
            </div>
          </div>
          <div
            class="bottom current"
          >
            0
          </div>
        </div>
        <p>
          hours
        </p>
      </div>
      <div
        class="sc-jSUZER czOrtm"
      >
        <div
          class="sc-gKPRtg bPprQx"
        >
          <div
            class="top next"
          >
            0
          </div>
          <div
            class="flipper"
          >
            <div
              class="bottom"
            >
              0
            </div>
            <div
              class="top"
            >
              0
            </div>
          </div>
          <div
            class="bottom current"
          >
            0
          </div>
        </div>
        <p>
          minutes
        </p>
      </div>
      <div
        class="sc-jSUZER czOrtm"
      >
        <div
          class="sc-gKPRtg bPprQx"
        >
          <div
            class="top next"
          >
            0
          </div>
          <div
            class="flipper"
          >
            <div
              class="bottom"
            >
              0
            </div>
            <div
              class="top"
            >
              0
            </div>
          </div>
          <div
            class="bottom current"
          >
            0
          </div>
        </div>
        <p>
          seconds
        </p>
      </div>
    </div>
    <p
      class="sc-dkrFOg eRASOh"
    >
      Your deadline has passed!
      <br />
      <br />
      Use the Reset button to update your deadline.
    </p>
    <button
      class="sc-eDvSVe zygpI"
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
